src/mach/imports.rs

```diff
@@ -195,9 +216,9 @@ impl<'a> BindInterpreter<'a> {
 	            // throwBadBindingAddress(address, segmentEndAddress, segmentIndex, start, end, p);
 	            // (this->*handler)(context, address, type, symbolName, symboFlags, addend, libraryOrdinal, "", &last);
 	            // address += sizeof(intptr_t);
+                    imports.push(Import::new(&bind_info, libs, segments, start_of_sequence));
                     let seg_offset = bind_info.seg_offset.wrapping_add(ctx.size() as u64);
                     bind_info.seg_offset = seg_offset;
-                    imports.push(Import::new(&bind_info, libs, segments));
```

Comment:
I don't think it was working. The comment says:

```rust
// we bind the old record, then increment bind info address for the next guy, plus the ptr offset *)
```

The comment matches [the behavior in `dyld`](https://github.com/opensource-apple/dyld/blob/master/src/ImageLoaderMachOCompressed.cpp#L999-L1004):

```c
case BIND_OPCODE_DO_BIND:
	if ( address >= segmentEndAddress ) 
		throwBadBindingAddress(address, segmentEndAddress, segmentIndex, start, end, p);
	(this->*handler)(context, address, type, symbolName, symboFlags, addend, libraryOrdinal, "", &last, false);
	address += sizeof(intptr_t);
	break;
```

Bind then mutate. The code had been doing the opposite.